api/app/controllers/spree/api/v2/storefront/products_controller.rb

```diff
@@ -0,0 +1,44 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        class ProductsController < ::Spree::Api::V2::BaseController
+          def show
+            render json: serialize_resource(resource), status: 200
+          end
+
+          private
+
+          def serialize_resource(resource)
+            dependencies[:resource_serializer].new(
+              resource,
+              include: %i[
+                variants
+                variants.images
+                option_types
+                option_types.option_values
+              ]
+            ).serializable_hash
+          end
+
+          def dependencies
+            {
+              resource_serializer: Spree::V2::Storefront::ProductSerializer
+            }
+          end
+
+          def resource
+            scope.find_by(slug: params[:id]) || scope.find(params[:id])
+          end
+
+          def scope
+            Spree::Product.includes(
```

Comment:
@kushniryb we definitely need more includes here - images and prices

Reply:
@damianlegawiec It's WIP but thanks for letting me know. Will you be able to deliver swagger doc related to this endpoint by tomorrow?